AndroidSDKCore/src/main/java/com/leanplum/monitoring/LeanplumCrashHandler.java

```diff
@@ -0,0 +1,41 @@
+package com.leanplum.monitoring;
+
+import android.content.Context;
+
+import com.leanplum.internal.Log;
+
+import java.lang.reflect.Constructor;
+
+public class LeanplumCrashHandler {
+  private static final String RAYGUN_CRASH_REPORTER_CLASS =
+      "com.leanplum.monitoring.internal.RaygunCrashReporter";
+  private static final LeanplumCrashHandler ourInstance = new LeanplumCrashHandler();
+
+  public static LeanplumCrashHandler getInstance() {
+    return ourInstance;
+  }
+
+  private LeanplumCrashReporting crashReporter;
+
+  private LeanplumCrashHandler() {}
+
+  public void setContext(Context context) {
+    try {
+      Class<?> clazz = Class.forName(RAYGUN_CRASH_REPORTER_CLASS);
```

Comment:
instead of using reflection to try to instantiate the class, we can just let the crash reporter register a factory, if it's loaded by the jvm.

so this CrashHandler can just expose a register method:
```
static void registerCrashReporterFactory(CrashReporterFactory crashReportFactory) {
 reporterFactory = crashReporterFactory;
}
```

then in setContext()

```
public void setContext(Context context) {
  if (reporterFactory != null) {
    crashReporter = crashReporterFactory.create(context);
  }
}
```

Reply:
Who calls `registerCrashReporterFactory` ? In my design, the module can just be added and won't need any additional code, hence reflection